common/src/main/java/com/wynntils/models/token/TokenModel.java

```diff
@@ -0,0 +1,252 @@
+    private final TimedSet<BakingTokenGatekeeper> bakingGatekeepers = new TimedSet<>(5, TimeUnit.SECONDS, true);
+    private final Map<Integer, TokenGatekeeper> invisibleGatekeepers = new HashMap<>();
+
+    public TokenModel() {
+        super(List.of());
+    }
+
+    public Stream<TokenGatekeeper> getGatekeepers() {
+        return activeGatekeepers.values().stream();
+    }
+
+    public CappedValue getCollected(TokenGatekeeper gatekeeper) {
+        TokenInventoryWatcher watcher = inventoryWatchers.get(gatekeeper);
+        // The null case should not happen, but just to be safe
+        int inventoryCount = watcher != null ? watcher.getTotalCount() : 0;
+
+        CappedValue deposited = gatekeeper.getDeposited();
+        int total = Math.min(deposited.current() + inventoryCount, deposited.max());
+        return new CappedValue(total, deposited.max());
+    }
+
+    @SubscribeEvent
+    public void onLabelChange(EntityLabelChangedEvent event) {
+        if (!(event.getEntity() instanceof ArmorStand)) return;
+
+        String name = event.getName();
+
+        Matcher typeMatcher = TYPE_PATTERN.matcher(name);
+        if (typeMatcher.matches()) {
+            String type = typeMatcher.group(1);
+
+            BakingTokenGatekeeper baking = getBaking(event.getEntity().position());
+            baking.type = type;
+            checkAndPromoteBaking();
+            return;
+        }
+
+        Matcher tokensMatcher = TOKEN_PATTERN.matcher(name);
+        if (tokensMatcher.matches()) {
+            CappedValue tokens =
+                    new CappedValue(Integer.parseInt(tokensMatcher.group(1)), Integer.parseInt(tokensMatcher.group(2)));
+
+            int id = event.getEntity().getId();
+            TokenGatekeeper gatekeeper = activeGatekeepers.get(id);
+            if (gatekeeper != null) {
+                // Update active gatekeeper
+                gatekeeper.setDeposited(tokens);
+                WynntilsMod.postEvent(new TokenGatekeeperEvent.Deposited(gatekeeper));
+                return;
+            }
+
+            TokenGatekeeper invisibleGatekeeper = invisibleGatekeepers.get(id);
+            if (invisibleGatekeeper != null) {
+                // We need to keep invisible gatekeepers up to date if they suddenly become visible
+                invisibleGatekeeper.setDeposited(tokens);
+                return;
+            }
+
+            // Create new baking gatekeeper
+            BakingTokenGatekeeper baking = getBaking(event.getEntity().position());
+            baking.value = tokens;
+            baking.valueEntityId = event.getEntity().getId();
+            checkAndPromoteBaking();
+            return;
+        }
+
+        // This does not provide any new information for the TokenGatekeeper, but it is a good
+        // extra check that we really match only on real gatekeepers
+        if (name.equals(VERIFICATION_STRING)) {
```

Comment:
```suggestion
        if (VERIFICATION_STRING.equals(name)) {
```

Reply:
Why??!? We know for sure name is not null. I distinctly unlike with a passion close to hate the Yoda-style comparison ("if 2 == var"), and can hardly even stand it even if it would save a null check...

Reply:
For some reason my eye prefered the suggestion but reading the code "name is X" is better than "X is name"